tools/lint/lint.py

```diff
@@ -59,9 +59,9 @@ def setup_logging(prefix=False):
 
 %s: %s"""
 
-def all_filesystem_paths(repo_root):
+def all_filesystem_paths(repo_root, extended_path):
```

Comment:
"`extended_path`" needs to be optional i.e. the signature should be

`all_filesystem_paths(repo_root, subdir=None):`

Also it should be relative  to `repo_root` so that in the function we do `root = os.path.join(repo_root, subdir) if subdir else repo_root` or something  similar.